casr/src/bin/casr-afl.rs

```diff
@@ -263,20 +264,27 @@ fn main() -> Result<()> {
     } else {
         std::cmp::max(1, num_cpus::get() / 2)
     };
-    let num_of_threads = jobs.min(crashes.len()).max(1);
+    let num_of_threads = jobs.min(crashes.len()).max(1) + 1;
     let custom_pool = rayon::ThreadPoolBuilder::new()
         .num_threads(num_of_threads)
         .build()
         .unwrap();
 
     // Generate CASR reports.
     info!("Generating CASR reports...");
-    info!("Using {} threads", num_of_threads);
-    custom_pool.install(|| {
-        crashes
-            .par_iter()
-            .try_for_each(|(_, crash)| crash.run_casr(output_dir.as_path(), timeout))
-    })?;
+    info!("Using {} threads", num_of_threads - 1);
+    let counter = RwLock::new(0_usize);
+    let total = crashes.len();
+    custom_pool.join(
+        || {
+            let _ = crashes.par_iter().try_for_each(|(_, crash)| {
+                crash.run_casr(output_dir.as_path(), timeout)?;
```

Comment:
Program may hang on error here and in casr-libfuzzer

Reply:
Done